### PR TITLE
Rebuild for libarrow 10.0.1

### PR DESCRIPTION
**Pull request by h-vetinari**
Following https://github.com/conda-forge/arrow-cpp-feedstock/pull/875; output `arrow-cpp` was renamed to `libarrow`.

### CODE REVIEW

**Comment by h-vetinari** [comment]
Not sure how I should pass CXXFLAGS in a way that they get picked up by `${R} CMD INSTALL`... Any tips @conda-forge/help-r?

**Comment by h-vetinari** [comment]
This also goes for windows, where we need to set C++17

File: recipe/build.sh [hunk 1]
```diff
@@ -11,4 +11,4 @@ if [[ "${target_platform}" == osx-* ]]; then
     export ARROW_R_CXXFLAGS="${ARROW_R_CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
 fi
 
-$R CMD INSTALL --build r/.
+${R} CMD INSTALL --build r/. ${R_ARGS}
```
Comment (h-vetinari) [inline recipe/build.sh:14]: this is necessary for cross-compilation to work, apparently

**Comment by h-vetinari** [comment]
@xhochy PTAL, I think this should be good to go 🙃

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Review by dbast — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**